activejob/lib/active_job/arguments.rb

```diff
@@ -158,11 +158,15 @@ def deserialize_hash(serialized_hash)
       end
 
       def serialize_hash_key(key)
-        case key
-        when *RESERVED_KEYS
+        if RESERVED_KEYS.include?(key)
```

Comment:
`Set#===` is an alias for `#include?` so you can do `case key when RESERVED_KEYS`.